React/nested-event-handling.md

```diff
@@ -124,6 +128,43 @@ Now let's take a look at an updated version of the diagram that we created in ou
 
 Note that we're passing in a new `isPresent` prop here. How could we modify our `Student` component to reflect this information?
 
+#### Removing State From `<Student />`
+
+Part of the concept of 'lifting state' is removing any state that is no longer relevant to the child element. 
+
+**Question:** What pieces of state still need to keep inside of the `Student`?
+
+<details>
+  <summary>Answer</summary>
+
+  Actually, none! While we _could_ make a working app that uses a class based component in the `Student` and the `StudentCollection`, it will be cleaner to read and update in the future if we turn the classical version of `Student` back into a functional component!
+</details>
+
+Ultimately, we can refactor the code to look like this:
+
+```javascript
+  // Student.js
+  const Student = (props) => {
+  return (
+    <section>
+      <h3>Student Component</h3>
+      <h4 className={attendanceClass}>Name {props.fullName} </h4>
+      <p>Email: {props.email} </p>
+    </section>
+  );}
+```
+
+```css
+   /*Student.css*/
+  .absent {
+    color: red;
+  }
+
+  .present {
```

Comment:
Do we have a naming convention for classes used for styling with React? I think the answer is no, but we probably should. Maybe something like, the outermost element of the component gets a class that is the component's name (`.student`), and then all styles reference both the component class and the modifier class (`.student .absent`).

Or just use BEM.

The reason I bring it up is, even though the CSS looks modular it does get applied to the entire site. I don't think we're likely to infect some other component using `.present` or `.absent` in this app, but it might be a risk for some of what the students are doing.

Reply:
Technically we do use BEM but frankly BEM is for people that don't understand how CSS selectors work.

I can go with the `.student .present` style, I follow the logic. It might be running back through the project series but * shrug *.